docs/_build/html/_sources/methods.rst.txt

```diff
@@ -0,0 +1,444 @@
+
+.. hint::
+
+  The ``set`` and ``get`` functions use the same syntax as the ``find`` function. They call it implicitly.
+
+.. _getGatingFunctions:
+
+getGatingFunctions
+^^^^^^^^^^^^^^^^^^
+
+::
+
+  [act ict tau_act tau_ict] = xolotl.getGatingFunctions('conductancePath')
+
+Returns the gating functions of a conductance as function handles. The function has four outputs.
+The activation steady state equation comes first, followed by the inactivation steady state (if any), then the activation time constant, and finally the inactivation time constant (if any).
+The argument is a string that specifies where in the file directory the conductance is specified.
+For example to find the ``NaV`` conductance from Liu *et al.* 1998, which is detailed in
+``.../xolotl/c++/conductances/liu/NaV.hpp`` ::
+
+  [m_inf, h_inf, tau_m, tau_h] = x.getGatingFunctions('liu/NaV')
+
+.. hint::
+
+  This is a static method of ``xolotl`` (i.e. ``xolotl.getGatingFunctions(...)`` is valid).
+
+.. _integrate:
+
+integrate
+^^^^^^^^^
+
+::
+
+  [V, Ca, cont_states, currents, syn_currents] = x.integrate
+
+Integrates the ``xolotl`` object. Returns the membrane potential, intracellular calcium, controller states, intrinsic currents, and synaptic currents as time series.
+
+The membrane potential is returned as matrix of ``time steps x compartments`` in the same order as the ``xolotl`` object's scalar representation (i.e. in the command window).
+The intracellular calcium is returned as a matrix of ``time steps x 2*compartments``. The first *n* time series are the intracellular calcium concentrations for each compartment, and the second *n* time series are the instantaneous calcium reversal potentials.
+Controller states are returned as a matrix of ``time steps x 2*controllers`` where the first column in each two-column pair is the controller state and the second is the state of the thing upon which the controller is operating.
+For a controller onto a maximal conductance, for instance, the first column is the controller state and the second is the maximal conductance.
+All other outputs are ``time series x XYZ`` where XYZ is the serialized list of those properties. For example, in a ``xolotl`` object with 2 compartments ``AB`` and ``BC``
+with two conductances each: ``NaV`` and ``Kd``, the intrinsic currents would return in the form of a ``time steps x 4`` matrix where the columns would be ordered:
+Kd current from AB, NaV current from AB, Kd current from BC, NaV current from BC.
+
+If ``integrate`` is passed one argument, it interprets this as a matrix of injected currents (in nA). For example ::
+
+  % applies 0.2 nA to each compartment
+  x.integrate(0.2)
+  % applies 0.2 nA to first compartment of three
+  x.integrate([0.2 0 0])
+  % applies a time-varying current to first compartment
+  % where time is a vector of length x.t_end / x.dt
+  x.integrate([0.2*sin(time); zeros(time); zeros(time)])
```

Comment:
don't specify sine()..., just say x.integrate(S), where S is a time varying current

Reply:
done